src/Configurator/CopyFromPackageConfigurator.php

```diff
@@ -53,52 +64,98 @@ private function copyFiles(array $manifest, string $from, string $to)
     private function removeFiles(array $manifest, string $from, string $to)
     {
         foreach ($manifest as $source => $target) {
-            $target = $this->options->expandTargetDir($target);
+            $targetPath = $this->concatenatePathParts([$to, $target]);
             if ('/' === substr($source, -1)) {
-                $this->removeFilesFromDir($from.'/'.$source, $to.'/'.$target);
+                $this->removeFilesFromDir(
+                    $this->concatenatePathParts([$from, $source]),
+                    $this->concatenatePathParts([$to, $target])
+                );
             } else {
-                @unlink($to.'/'.$target);
+                @unlink($targetPath);
+                $this->write(sprintf(
+                    'Removed file <fg=green>"%s"</>',
+                    $this->relativizePath($targetPath)
+                ));
             }
         }
     }
 
-    private function copyDir(string $source, string $target)
+    private function copyDir(string $source, string $target): void
     {
         if (!is_dir($target)) {
             mkdir($target, 0777, true);
         }
 
-        $iterator = new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($source, \RecursiveDirectoryIterator::SKIP_DOTS), \RecursiveIteratorIterator::SELF_FIRST);
+        $iterator = new \RecursiveIteratorIterator(
+            new \RecursiveDirectoryIterator($source, \RecursiveDirectoryIterator::SKIP_DOTS),
+            \RecursiveIteratorIterator::SELF_FIRST
+        );
         foreach ($iterator as $item) {
+            $targetPath = $this->concatenatePathParts([$target, $iterator->getSubPathName()]);
             if ($item->isDir()) {
-                if (!is_dir($new = $target.'/'.$iterator->getSubPathName())) {
-                    mkdir($new);
+                if (!is_dir($targetPath)) {
+                    mkdir($targetPath);
+                    $this->write(sprintf(
+                        'Created directory <fg=green>"%s"</>',
+                        $this->relativizePath($targetPath)
+                    ));
                 }
-            } elseif (!file_exists($target.'/'.$iterator->getSubPathName())) {
-                $this->copyFile($item, $target.'/'.$iterator->getSubPathName());
+            } elseif (!file_exists($targetPath)) {
+                $this->copyFile($item, $targetPath);
             }
         }
     }
 
-    public function copyFile(string $source, string $target)
+    public function copyFile(string $source, string $target): void
```

Comment:
Should this method really be public? I have found no usage for it outside of the this class.

Reply:
it is now, so better not break it - but "void" must be removed: Flex is 7.0-compatible

Reply:
Yes I have already removed it, this is just an old diff.